modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/inputs/IntegerInput.tsx

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {ClaySelectWithOption} from '@clayui/form';
+import React from 'react';
+
+
+interface Props {
+    disabled?: boolean;
+    onChange: (payload: {value: string}) => void;
+    options: [
+        {
+            disabled: boolean,
+            label: string,
+            value: string,
+        }
+    ]
+
+    value?: number | string;
+}
+
+function IntegerInput({
+                          disabled,
+                          onChange,
+                          options,
+                          value
+                      }: Props) {
+
+
+    const handleIntegerChange = (event: any) => {
```

Comment:
We should avoid using any, it just removes any type checking and accepts any kind of values. There are many builtin types but event handlers like this one. For example for change events you can use the generic `Event` interface :smile:

Reply:
Got it, thank u!